File: src/front-end/src/services/invoice.service.js
import {BASE_API_URL} from "../common/Constants";
import axios from "axios";
import {authHeader} from "./base.service";


const API_URL = BASE_API_URL + '/gateway/invoices';


class InvoiceService {
    createNewInvoice(invoice) {
        return axios.post(API_URL, invoice, {headers: authHeader()});
    }

    getAllInvoices() {
        return axios.get(API_URL, {headers: authHeader()});
    }

    getInvoiceByReference(invoice) {
        return axios.get(API_URL+ '/reference' + invoice.reference, {headers: authHeader()});
    }

    payInvoice(reference) {
        return axios({
                method: 'put',
                url: API_URL + '/' + reference + '/pay',
                headers: authHeader()
            }
        )
    }

    cancelInvoice(invoice) {
        return axios.delete(API_URL+ '/reference' + invoice.reference + '/cancel', {headers: authHeader()});
    }

}


export default new InvoiceService();

File: finance-invoice/src/main/java/com/lolucode/financeInvoice/service/InvoiceService.java
package com.lolucode.financeInvoice.service;

import com.lolucode.financeInvoice.dto.InvoiceDto;
import com.lolucode.financeInvoice.model.Invoice;

import java.util.List;

public interface InvoiceService {

    InvoiceDto createNewInvoice(Invoice invoice);

    List<InvoiceDto> viewAllInvoicesOfUser(Long userId);

    Invoice getInvoiceByReference(String reference);

    Invoice pay(String reference);

    Invoice cancel(String reference);

    Invoice processPayment(String reference);

    Invoice payInvoice(Invoice invoice);

    Invoice processInvoice(Invoice invoice);
}
